var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -576,54 +578,118 @@ def headers(self):
+                   '-snes_monitor']
+        purpose_str = 'test: run ex3k.kokkos example with +kokkos'
+
+        self.run_test(runexe,
+                      options=runopt + testexe,
+                      purpose=purpose_str,
+                      work_dir=w_dir)
 
     def test(self):
         # solve Poisson equation in 2D to make sure nothing is broken:
-        spec = self.spec
         env['PETSC_DIR'] = self.prefix
         env['PETSC_ARCH'] = ''
-        if ('+mpi' in spec):
-            runexe = Executable(join_path(spec['mpi'].prefix.bin,
+        if ('+mpi' in self.spec):
+            runexe = Executable(join_path(self.spec['mpi'].prefix.bin,
                                           'mpiexec')).command
-            runopt = ['-n', '4']
         else:
-            runexe = Executable(join_path(self.prefix,
-                                          'lib/petsc/bin/petsc-mpiexec.uni')).command
-            runopt = ['-n', '1']
-        w_dir = join_path(self.install_test_root, 'src/ksp/ksp/tutorials')
-        with working_dir(w_dir):
-            testexe = ['ex50', '-da_grid_x', '4', '-da_grid_y', '4']
-            testdict = {
-                None: [],
-                '+superlu-dist':
-                ['-pc_type', 'lu', '-pc_factor_mat_solver_type', 'superlu_dist'],
-                '+mumps':
-                ['-pc_type', 'lu', '-pc_factor_mat_solver_type', 'mumps'],
-                '+hypre':
-                ['-pc_type', 'hypre', '-pc_hypre_type', 'boomeramg'],
-                '+mkl-pardiso':
-                ['-pc_type', 'lu', '-pc_factor_mat_solver_type', 'mkl_pardiso'],
-            }
-            make('ex50', parallel=False)
-            for feature, featureopt in testdict.items():
-                if not feature or feature in spec:
-                    self.run_test(runexe, runopt + testexe + featureopt)
-            if '+cuda' in spec:
-                make('ex7', parallel=False)
-                testexe = ['ex7', '-mat_type', 'aijcusparse',
-                           '-sub_pc_factor_mat_solver_type', 'cusparse',
-                           '-sub_ksp_type', 'preonly', '-sub_pc_type', 'ilu',
-                           '-use_gpu_aware_mpi', '0']
-                self.run_test(runexe, runopt + testexe)
-            make('clean', parallel=False)
-        w_dir = join_path(self.install_test_root, 'src/snes/tutorials')
-        with working_dir(w_dir):
-            if '+kokkos' in spec:
-                make('ex3k', parallel=False)
-                testexe = ['ex3k', '-view_initial', '-dm_vec_type', 'kokkos',
-                           '-dm_mat_type', 'aijkokkos', '-use_gpu_aware_mpi', '0',
-                           '-snes_monitor']
-                self.run_test(runexe, runopt + testexe)
-            make('clean', parallel=False)
+            runexe = Executable(join_path(self.prefix.lib.petsc.bin,
+                                          'petsc-mpiexec.uni')).command
```

Comment:
There is no reason/need to create the `Executable` of the `runexe` if it is being passed to `self.run_test()` since the method will automatically do that for the provided argument.